### DISCUSSION TITLE
How to run one-time job at specific time in future?

### DISCUSSION THREAD

**Discussion by ProteanCode** [laravel/framework#44197] [Q&A]
I am trying to do something like that

```
public function playground(Schedule $schedule)
{
    $job = new SpecificDateTimeJob();

    $schedule->job($job)->at(Carbon::now()->addHours(5));
}
```
Where playground is some controller method

But this does not put a task in jobs table

I must also be able to update that job because its start is equal to a database field value which may change

**Comment by henzeb** [comment]
Schedule is cronjob. You need to dispatch a job with a delay.

https://laravel.com/docs/9.x/queues#delayed-dispatching

**Reply by ProteanCode** [reply]
> Schedule is cronjob. You need to dispatch a job with a delay.
> 
> https://laravel.com/docs/9.x/queues#delayed-dispatching

Ah yesss, I totally forgot about that. Thank you

Now I am looking for a way to update the delayed value once stored in database. I have tried to work with unique job, but this didn't work.
I have a case where user can update the end date of an event, and this should update the delayed job to that specific date.

Calling `delayed` multiple times for unique job seems to do not work (sadly)

**Comment by syamsoul** [comment]
hi, @ProteanCode ...

i also have problem like you.. then, i decided to develop my own laravel package that can ease your life...

https://github.com/syamsoul/laravel-action-delay

just install via `composer require syamsoul/laravel-action-delay`

and run `php artisan souldoit:action-delay`

the system will ask all the things that you need, for example:

```bash
 What action you want to delay? [Laravel Jobs]:
  [1] Laravel Jobs
  [2] Database Query
  [3] PHP Code
  [4] External Process
 > 2

 Enter MySQL query:
 > UPDATE `variables` SET `_value`='false' WHERE `_key`='maintainance_mode_enabled'

 What time to execute (in UTC time, format:Y-m-d H:i:s):
 > 2024-06-01 08:30:35
 ```
 
 
 another example
 
 ```bash
  What action you want to delay? [Laravel Jobs]:
  [1] Laravel Jobs
  [2] Database Query
  [3] PHP Code
  [4] External Process
 > 3

 Enter PHP code:
 > \App\Models\Variable::where('_key', 'maintainance_mode_enabled')->update(['_value' => 'false']); \App\Models\Variable::where('_key', 'new_feature_enabled')->update(['_value' => 'true']);

 What time to execute (in UTC time, format:Y-m-d H:i:s):
 > 2024-06-01 08:30:35
 ```
